src/ray/gcs/gcs_server/gcs_init_data.h

```diff
@@ -89,7 +89,7 @@ class GcsInitData {
 
  protected:
   /// The gcs table storage.
-  std::shared_ptr<gcs::GcsTableStorage> gcs_table_storage_;
+  gcs::GcsTableStorage &gcs_table_storage_;
```

Comment:
I see reference and pointer are both used. What's the rule you are following?

Reply:
- Prefer to use reference as data member, since it guarantee non-null;
- Have to use pointer in some places due to our testing implementation.